app/models/aggregation.rb

```diff
@@ -14,8 +14,11 @@ class Aggregation < ActiveRecord::Base
   validate :aggregation, :workflow_version_present
 
   def self.scope_for(action, user, opts={})
-    return super unless (action == :show || action == :index) && !user.is_admin?
-    joins(:workflow).merge(Workflow.scope_for(:update, user, opts))
+    if (action == :show || action == :index) && !user.is_admin?
+      joins(:workflow).merge(Workflow.scope_for(:update, user, opts))
```

Comment:
And you also wouldn't be restricted to explicitly requested public workflows, but that doesn't seem like a thing we do elsewhere.